modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1281,6 +1281,83 @@ CV__DNN_INLINE_NS_BEGIN
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+         CV_WRAP TextRecognitionModel(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from deep learning network.
+          * @param[in] network Net object.
+          */
+         CV_WRAP TextRecognitionModel(const Net& network);
+
+         /**
+          * @brief Given the @p input frame, create input blob, run net and return recognition result.
+          * @param[in] frame: The input image.
+          * @param[in] decodeType: The decoding method of translating the network output into string.
+          * @param[in] vocabulary: The vocabulary for decoding.
+          * @param[out] results: A set of text recognition results.
+          * @param[in] roiPolygons: A set of text detection results.
+          */
+         CV_WRAP void recognize(InputArray frame, String decodeType, const std::vector<String> & vocabulary,
+                                CV_OUT std::vector<String> & results,
+                                const std::vector<std::vector<Point>> & roiPolygons = std::vector<std::vector<Point>>());
+
+     };
+
+     /** @brief This class represents high-level API for text detection networks.
+      *
+      * TextDetectionModel allows to set params for preprocessing input image.
+      * TextDetectionModel creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return detection results.
+      * For TextDetectionModel, "DB" is supported.
+      */
+     class CV_EXPORTS_W TextDetectionModel : public Model
+     {
+     public:
+         /**
+          * @brief Create text detection model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+         CV_WRAP TextDetectionModel(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from deep learning network.
+          * @param[in] network Net object.
+          */
+         CV_WRAP TextDetectionModel(const Net& network);
+
+         /**
+          * @brief Given the @p input frame, create input blob, run net and return detection results.
+          * @param[in] frame: The input image.
+          * @param[out] results: A set of text detection results.
+          * @param[in] outputType: The output type: {0: multi-oriented quadrilateral; 1: polygon}
+          * @param[in] binThresh: The threshold of the binary map.
+          * @param[in] polyThresh: The threshold of text polygons.
+          * @param[in] unclipRatio: The unclip ratio of the detected text region.
+          * @param[in] maxCandidates: The max number of the output polygons.
```

Comment:
Why we need to limit this number?

Reply:
We can get good detection results with these parameters, for more information you can refer to the [paper](https://arxiv.org/abs/1911.08947) and [code](https://github.com/MhLiao/DB).

Reply:
In some difficult cases, the output map of the network is full of small noise, and `maxCandidates` can avoid wasting inference time.